src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -154,10 +154,10 @@ static CellIdx getLeastUsedCell(Random &rng, UInt column, //TODO remove static m
   const CellIdx start = column * cellsPerColumn;
   const CellIdx end = start + cellsPerColumn;
 
-  UInt32 minNumSegments = UINT_MAX;
-  UInt32 numTiedCells = 0;
+  CellIdx minNumSegments = UINT_MAX;
```

Comment:
here also now `num_limit<CelldIdx> `, as a programmer might change the type and overflow here

Reply:
ok..try to implement this one first 
CC @dkeeney